graphistry/umap_utils.py

```diff
@@ -411,8 +411,15 @@ def _process_umap(
         print('** Fitting UMAP') if verbose else None
         res = res.umap_lazy_init(res, verbose=verbose, **umap_kwargs_pure)
         
+        self.datetime_columns = X_.select_dtypes(
+            include=["datetime", "datetimetz"]
+        ).columns.to_list()
+        
+        self.R_ = X_[self.datetime_columns]
+        X_ = X_.drop(columns=self.datetime_columns)
+        
         emb = res._umap_fit_transform(X_, y_, verbose=verbose)
-        res._xy = emb
+        res._xy = emb.join(self.R_)
         return res
```

Comment:
@silkspace  what should happen here? it sounds like umap does not support df cols, so is this the right workaround?

@dcolinmorgan is this the dt issue you're talking about, or there's another?

Reply:
i thought it was a cudf/cupy issue for some time, but actually i think tested and it seems dirty_cat datetime 'passthrough' just does not work, so need to fix that-- should resolve issues cody and I been seeing using cucat w/dt

Reply:
[they have wanted to encode such data](https://skrub-data.org/stable/auto_examples/03_datetime_encoder.html#one-liner-with-the-tablevectorizer), but doesnt make sense for us -- they provide a passthrough [but it only works on numeric ](https://github.com/skrub-data/skrub/blob/main/skrub/_table_vectorizer.py#L847C1-L848C1) -- top priority to sort this out

Reply:
ok -- **working** on [g@dtpass](https://github.com/graphistry/pygraphistry/tree/dtpass) and [cu_cat@DT5](https://github.com/graphistry/cu-cat/tree/DT5) -- will merge

Reply:
Fwiw, do we need to update to track latest dirty car, as it has been awhile?

(Maybe do as a follow-on PR after this lands?)